cloudinit/distros/photon.py

```diff
@@ -0,0 +1,380 @@
+from cloudinit import subp
+from cloudinit import distros
+from cloudinit import helpers
+from cloudinit import log as logging
+from cloudinit.distros import net_util
+from cloudinit.settings import PER_INSTANCE
+from cloudinit.distros import rhel_util as rhutil
+from cloudinit.net.network_state import mask_to_net_prefix
+from cloudinit.distros.parsers.hostname import HostnameConf
+
+LOG = logging.getLogger(__name__)
+
+def network_file_name(dirname, dev_name):
+    network_file_pattern = '*' + dev_name + '.network*'
+    for node in os.listdir(dirname):
+        if fnmatch.fnmatch(node, network_file_pattern):
+            node_fullpath = os.path.join(dirname, node)
+            util.del_file(node_fullpath)
+            return dirname + node
+
+    return dirname + '10-' + dev_name + '.network'
+
+
+class Distro(distros.Distro):
+    hostname_conf_fn = '/etc/hostname'
+    locale_conf_fn = '/etc/default/locale'
+    network_conf_dir = '/etc/systemd/network/'
+    systemd_locale_conf_fn = '/etc/locale.conf'
+    resolve_conf_fn = '/etc/systemd/resolved.conf'
+
+    renderer_configs = {
+        'networkd': {
+            'resolv_conf_fn': resolve_conf_fn,
+            'network_conf_dir': network_conf_dir,
+            'network_fname_fptr': network_file_name,
+        }
+    }
+
+    # Should be fqdn if we can use it
+    prefer_fqdn = True
+
+    def __init__(self, name, cfg, paths):
+        distros.Distro.__init__(self, name, cfg, paths)
+        # This will be used to restrict certain
+        # calls from repeatly happening (when they
+        # should only happen say once per instance...)
+        self._runner = helpers.Runners(paths)
+        self.osfamily = 'photon'
+        self.init_cmd = ['systemctl']
+
+    def apply_locale(self, locale, out_fn=None):
+        if self.uses_systemd():
+            if not out_fn:
+                out_fn = self.systemd_locale_conf_fn
+        else:
+            if not out_fn:
+                out_fn = self.locale_conf_fn
+        locale_cfg = {
+            'LANG': locale,
+        }
+        rhutil.update_sysconfig_file(out_fn, locale_cfg)
+
+        # rhutil will modify /etc/locale.conf
+        # For locale change to take effect, reboot is needed or we can restart
+        # systemd-localed. This is equivalent of localectl
+        cmd = ['systemctl', 'restart', 'systemd-localed']
+        LOG.debug('Attempting to restart localed using command %s', cmd)
+        try:
+            (_out, err) = subp.subp(cmd)
+            if len(err):
+                LOG.warning('Running %s resulted in stderr output: %s',
+                            cmd, err)
+        except subp.ProcessExecutionError:
+            util.logexc(LOG, 'Restart of localed using command %s failed', cmd)
+
+    def install_packages(self, pkglist):
+        # self.update_package_sources()
+        self.package_command('install', pkgs=pkglist)
+
+    def _write_network_config(self, netconfig):
+        if isinstance(netconfig, str):
```

Comment:
We have some internal dependency on this function, down the line I will make it right. As of now, let it be.

Reply:
Ok, I'll hold off from reviewing the networking stuff in this file for now. Let me know when it's the way you want it.

Reply:
Sure, once this PR gets merged I will send incremental patches. I have touched too many files already, so I don't want to disrupt this at the moment.